ethereum/contracts/relayer/relayProvider/RelayProvider.sol

```diff
@@ -6,28 +6,40 @@ pragma solidity ^0.8.19;
 import "./RelayProviderGovernance.sol";
 import "./RelayProviderStructs.sol";
 import "../../interfaces/relayer/IRelayProvider.sol";
+import "../../interfaces/relayer/TypedUnits.sol";
 
 contract RelayProvider is RelayProviderGovernance, IRelayProvider {
+    using WeiLib for Wei;
+    using GasLib for Gas;
+    using GasPriceLib for GasPrice;
+    using WeiPriceLib for WeiPrice;
+
     error CallerNotApproved(address msgSender);
 
     //Returns the delivery overhead fee required to deliver a message to the target chain, denominated in this chain's wei.
     function quoteDeliveryOverhead(uint16 targetChainId)
         public
         view
         override
-        returns (uint256 nativePriceQuote)
+        returns (Wei nativePriceQuote)
     {
-        uint256 targetFees = uint256(deliverGasOverhead(targetChainId)) * gasPrice(targetChainId);
-        return quoteAssetConversion(targetChainId, targetFees, chainId());
+        Gas overhead = deliverGasOverhead(targetChainId);
+        Wei targetFees = overhead.toWei(gasPrice(targetChainId));
+        Wei result = quoteAssetConversion(targetChainId, targetFees, chainId());
+        require(result.unwrap() <= type(uint128).max, "Overflow");
```

Comment:
Yeah, the only real users of this will be xlabs and we'll be much more free to upgrade/polish these impls over time. IMO the relayProvider impl itself is quite verbose atm anyway and could use some simplification + refactor love anyway. 

(^ AKA the excuses I tell myself so I can sleep at night and still merge prs)